src/Devito.jl

```diff
@@ -775,6 +775,28 @@ function Base.getindex(x::Union{TimeFunction,Function},args...)
    return py"indexobj"(x,reverse(args)...)
 end
 
-export DiscreteFunction, Grid, Function, SpaceDimension, SparseTimeFunction, SteppingDimension, TimeDimension, TimeFunction, apply, backward, configuration, configuration!, coordinates, data, data_allocated, data_with_halo, data_with_inhalo, dimensions, dx, dy, dz, extent, forward, grid, inject, interpolate, localindices, localindices_with_halo, localindices_with_inhalo, localsize, size_with_halo, spacing, spacing_map, step
+
+"""
+Create a subdomain by passing a list of instructions for each dimension
+Example:
+instructions = [("left",2),("middle",3,3)]
+SubDomin("subdomainName",instructions)
+"""
+function SubDomain(name, instructions)
+    instructions = reverse(instructions)
+    @pydef mutable struct subdomain <: devito.SubDomain
+        function __init__(self, name, instructions )
+            self.name = name
+            self.instructions = instructions
+        end
+        function define(self, dimensions)
+            dim =  Dict(dimensions[1] => self.instructions[1] , dimensions[2] => self.instructions[2])
```

Comment:
Doesn't this assume two dimensions.  Is there a way to generalize to n dimensions?